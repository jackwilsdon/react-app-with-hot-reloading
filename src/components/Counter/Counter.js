import './Counter.css';
import { decrement, increment } from '../../actions';
import PropTypes from 'prop-types';
import React from 'react';
import { connect } from 'react-redux';
import { getValue } from '../../selectors/counter';

export const Counter = ({ onSubtract, onAdd, value }) => (
  <div className="Counter">
    <button className="Counter__button" onClick={() => onSubtract()}>
      -
    </button>
    <div className="Counter__value">{value}</div>
    <button className="Counter__button" onClick={() => onAdd()}>
      +
    </button>
  </div>
);

Counter.propTypes = {
  onSubtract: PropTypes.func.isRequired,
  onAdd: PropTypes.func.isRequired,
  value: PropTypes.any,
};

Counter.defaultProps = {
  value: null,
};

const mapStateToProps = state => ({
  value: getValue(state),
});

const mapDispatchToProps = {
  onAdd: increment,
  onSubtract: decrement,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Counter);
