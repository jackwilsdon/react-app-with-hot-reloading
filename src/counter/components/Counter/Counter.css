.Counter {
  display: inline-flex;
}

.Counter__button {
  border: none;
  background-color: #222;
  color: #fff;
  width: 50px;
  height: 50px;
  cursor: pointer;
}

.Counter__button:hover,
.Counter__button:focus {
  background-color: #555;
  outline: none;
}

.Counter__button:active {
  background-color: #222;
}

.Counter__button::-moz-focus-inner {
  border: none;
}

.Counter__value {
  min-width: 50px;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 0 10px;
  box-sizing: border-box;
}
