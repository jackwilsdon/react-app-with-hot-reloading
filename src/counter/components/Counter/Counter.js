import { decrement, getValue, increment } from '../../ducks';
import React from 'react';
import { connect } from 'react-redux';
import './Counter.css';

const Counter = ({ onSubtract, onAdd, value }) => (
  <div className="Counter">
    <button className="Counter__button" onClick={() => onSubtract()}>
      -
    </button>
    <div className="Counter__value">{value}</div>
    <button className="Counter__button" onClick={() => onAdd()}>
      +
    </button>
  </div>
);

const mapStateToProps = state => ({
  value: getValue(state),
});

const mapDispatchToProps = {
  onAdd: increment,
  onSubtract: decrement,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Counter);
