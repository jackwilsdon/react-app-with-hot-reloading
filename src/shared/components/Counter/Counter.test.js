import { Counter } from './Counter';
import React from 'react';
import noop from 'lodash/noop';
import { shallow } from 'enzyme';

it('renders correctly', () => {
  const tree = shallow(
    <Counter onAdd={noop} onSubtract={noop} value={12345} />,
  );

  expect(tree).toMatchSnapshot();
});
