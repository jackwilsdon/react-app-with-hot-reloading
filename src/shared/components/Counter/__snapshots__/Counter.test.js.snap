// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="Counter"
>
  <button
    className="Counter__button"
    onClick={[Function]}
  >
    -
  </button>
  <div
    className="Counter__value"
  >
    12345
  </div>
  <button
    className="Counter__button"
    onClick={[Function]}
  >
    +
  </button>
</div>
`;
